### PR TITLE
Removed python codes and added spotinst resource to `create account` and `set credential` along with google TF resource to create a spot account and connect to GCP project.

### PR DESCRIPTION
**Pull request by sharadkesarwani**
Removed python codes and added spotinst resource to `create account` and `set credential` along with google TF resource to create a spot account and connect to GCP project.

# Jira Ticket
Ref : https://spotinst.atlassian.net/browse/SPOTAUT-19459

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,92 +1,75 @@
...
+    service_account_id = google_service_account.spotserviceaccount.name
 }
 
 resource "google_project_iam_binding" "spot-account-iam" {
     project = var.project
-    role = google_project_iam_custom_role.SpotRole.name
+    role    = google_project_iam_custom_role.SpotRole.name
     members = [
-        "serviceAccount:spot-${random_id.role.hex}@${var.project}.iam.gserviceaccount.com",
+        google_service_account.spotserviceaccount.member
```
Comment (anuragsharma-123) [inline main.tf:43]: Is this member's change correct?

Comment (sharadkesarwani) [reply main.tf:43]: Yes, followed the same ways as used in gcp-connect module.

File: main.tf [hunk 2]
```diff
@@ -1,92 +1,75 @@
...
-        local.cmd,
-        "get",
-        "--filter=name=${var.project}"
+resource "google_project_iam_binding" "service-account-user-iam" {
+    project = var.project
+    role    = "roles/iam.serviceAccountUser"
+    members = [
+        google_service_account.spotserviceaccount.member
     ]
 }
```
Comment (chandra1-n) [inline main.tf]: add comment what set credentials doing

File: main.tf [hunk 3]
```diff
@@ -1,92 +1,75 @@
...
+    }
+    account_id = spotinst_account.spot_acct.id
+    type = "service_account"
+    project_id = jsondecode(base64decode(google_service_account_key.key.private_key))["project_id"]
+    private_key_id = jsondecode(base64decode(google_service_account_key.key.private_key))["private_key_id"]
+    private_key = jsondecode(base64decode(google_service_account_key.key.private_key))["private_key"]
+    client_email = jsondecode(base64decode(google_service_account_key.key.private_key))["client_email"]
+    client_id = jsondecode(base64decode(google_service_account_key.key.private_key))["client_id"]
+    auth_uri = jsondecode(base64decode(google_service_account_key.key.private_key))["auth_uri"]
+    token_uri = jsondecode(base64decode(google_service_account_key.key.private_key))["token_uri"]
```
Comment (chandra1-n) [inline main.tf:64]: re-check this if it is copy paste issue

**Review by chandra1-n — CHANGES_REQUESTED** [review]
please fix the review comments

File: main.tf (hunk 2 above)
Comment (sharadkesarwani) [reply main.tf]: added comment

**Review by chandra1-n — APPROVED** [review]

**Pull request merged**